src/libs/MoneyRequestUtils.ts

```diff
@@ -38,15 +38,21 @@ function addLeadingZero(amount: string): string {
 }
 
 /**
- * Check if amount is a decimal up to 3 digits
+ * Get amount regex string
+ */
+function amountRegex(decimals: number, amountMaxLength: number = CONST.IOU.AMOUNT_MAX_LENGTH): string {
+    return decimals === 0
+        ? `^\\d{0,${amountMaxLength}}$` // Don't allow decimal point if decimals === 0
+        : `^\\d{0,${amountMaxLength}}(\\.\\d{0,${decimals}})?$`; // Allow the decimal point and the desired number of digits after the point
```

Comment:
How does work with internationalization where `,` is used instead of a `.`?

Reply:
Oh great catch! I think I broke that 😅 On main we replace the commas with periods but with the regex prop you won't be even allowed to insert a comma. The problem is due to the fact that what the user is allowed to enter and what we consider a valid input are two different things (because we format the user input).

i believe same bug occurs with entering `.45`
- `.45` is not a valid input
-  but the user can still enter `.45` because we format it to `0.45`

Reply:
I'm going to bring this back to draft to see how we can handle this. We can have 2 regexes (one of user input validation and the other for formatted text validation) but I don't really like that option. Will look for better options

Reply:
I just realized that the second problem won't occur because the modified regex already allows for `.45`. To keep this simple I added the comma to the regex amount so we can insert it and we still replace it with a period.


https://github.com/user-attachments/assets/1a8a7da1-6653-4bca-8553-dca64ff9617f